### ISSUE TITLE
bug report

### ISSUE THREAD

**Issue by nevakrien** [llvm/llvm-project#103483]
was told by clang to submit a bug report... very simple program. If this is a knowen issue or I did something wrong then I am sorry for the false report

```
define void @quick_sort(i32* %arr,i64 %len) {
	entry:
	%end_p = getelementptr i32*,i32* %arr,i64 %len
	%first = getelementptr i32*,i32* %arr, i32 1
	%pivot = load i32,i32* %arr


	%empty = icmp sle i64 %len, 0
	br i1 %empty, label %end_empty, label %loop
	
	loop:
		%p =  phi i32* [ %first, %entry ], [ %next_p, %loop ]
		%next_p = getelementptr i32*,i32* %p,i32 1
		%done = icmp eq i32* %next_p, %end_p


		; %insert_left = phi i32* [ %first, %entry ], [ %next_left, %loop ]
		; %insert_right = phi i32* [ %first, %entry ], [ %next_right, %loop ]

		%elem = load i32, i32* %p
		%biger = icmp sgt i32 %elem , %pivot

		br i1 %biger, label %right,label %left
	left:
		br i1 %done, label %recurse, label %loop

	right:
		br i1 %done, label %recurse, label %loop
	
	recurse:
	ret void

	end_empty:
	ret void
}
```
```bash
(base) user@user-System-Product-Name:~/Desktop/assembly/first_llvm$ clang quick_sort.ll -S -masm=intel
warning: overriding the module target triple with x86_64-unknown-linux-gnu [-Woverride-module]
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: clang quick_sort.ll -S -masm=intel
1.	Code generation
2.	Running pass 'Function Pass Manager' on module 'quick_sort.ll'.
3.	Running pass 'X86 DAG->DAG Instruction Selection' on function '@quick_sort'
 #0 0x00005572b88d7af8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x6161af8)
 #1 0x00005572b88d588e llvm::sys::RunSignalHandlers() (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x615f88e)
 #2 0x00005572b88d6f81 llvm::sys::CleanupOnSignal(unsigned long) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x6160f81)
 #3 0x00005572b884a529 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #4 0x00007eb155842520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #5 0x00005572b979a1e9 llvm::FastISel::handlePHINodesInSuccessorBlocks(llvm::BasicBlock const*) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x70241e9)
 #6 0x00005572b979990a llvm::FastISel::selectInstruction(llvm::Instruction const*) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x702390a)
 #7 0x00005572b986016f llvm::SelectionDAGISel::SelectAllBasicBlocks(llvm::Function const&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x70ea16f)
 #8 0x00005572b985e78c llvm::SelectionDAGISel::runOnMachineFunction(llvm::MachineFunction&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x70e878c)
 #9 0x00005572b7b5491f (anonymous namespace)::X86DAGToDAGISel::runOnMachineFunction(llvm::MachineFunction&) X86ISelDAGToDAG.cpp:0:0
#10 0x00005572b7fc95c7 llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x58535c7)
#11 0x00005572b83ecc76 llvm::FPPassManager::runOnFunction(llvm::Function&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x5c76c76)
#12 0x00005572b83f3da3 llvm::FPPassManager::runOnModule(llvm::Module&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x5c7dda3)
#13 0x00005572b83ed910 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x5c77910)
#14 0x00005572b8f7965f clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>, clang::BackendConsumer*) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x680365f)
#15 0x00005572b8f9b3f9 clang::CodeGenAction::ExecuteAction() (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x68253f9)
#16 0x00005572b9345817 clang::FrontendAction::Execute() (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x6bcf817)
#17 0x00005572b92bdd54 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x6b47d54)
#18 0x00005572b940ad7e clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x6c94d7e)
#19 0x00005572b692eca7 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x41b8ca7)
#20 0x00005572b692ba51 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&, llvm::ToolContext const&) driver.cpp:0:0
#21 0x00005572b911b889 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const::$_0>(long) Job.cpp:0:0
#22 0x00005572b884a30c llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x60d430c)
#23 0x00005572b911b4f6 clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x69a54f6)
#24 0x00005572b90dd314 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x6967314)
#25 0x00005572b90dd7ae clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x69677ae)
#26 0x00005572b90fbf9f clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x6985f9f)
#27 0x00005572b692b09b clang_main(int, char**, llvm::ToolContext const&) (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x41b509b)
#28 0x00005572b693a7e7 main (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x41c47e7)
#29 0x00007eb155829d90 __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
#30 0x00007eb155829e40 call_init ./csu/../csu/libc-start.c:128:20
#31 0x00007eb155829e40 __libc_start_main ./csu/../csu/libc-start.c:379:5
#32 0x00005572b6927dea _start (/home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-18+0x41b1dea)
clang: error: clang frontend command failed with exit code 139 (use -v to see invocation)
clang version 18.1.8
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/user/repo_packages/clang_new/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin
clang: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs.
(base) user@user-System-Product-Name:~/Desktop/assembly/first_llvm$ 
```

**Comment by asl** [comment]
```llvm
	loop:
		%p =  phi i32* [ %first, %entry ], [ %next_p, %loop ]
```

This is clearly invalid. Phi node should contain entry for each predecessor BB. However, `%loop` does not branch to itself. Overall, I would suggest you to run IR verifier on your LLVM IR before feeding to clang. It assumes the input is correct.
